src/traces/histogram/calc.js

```diff
@@ -180,10 +157,181 @@ module.exports = function calc(gd, trace) {
     return cd;
 };
 
+/*
+ * calcAllAutoBins: we want all histograms on the same axes to share bin specs
+ * if they're grouped or stacked. If the user has explicitly specified differing
+ * bin specs, there's nothing we can do, but if possible we will try to use the
+ * smallest bins of any of the auto values for all histograms grouped/stacked
+ * together.
+ */
+function calcAllAutoBins(gd, trace, pa, maindata) {
+    var binAttr = maindata + 'bins';
+
+    // all but the first trace in this group has already been marked finished
+    // clear this flag, so next time we run calc we will run autobin again
+    if(trace._autoBinFinished) {
```

Comment:
Maybe we should move auto-bins computations to `histograms/set_positions` (which loops over all traces of a given trace type) to avoid having to use `_` flags like this one?